clang/include/clang/Basic/Attr.td

```diff
@@ -4588,6 +4588,35 @@ def HLSLResourceBinding: InheritableAttr {
   let LangOpts = [HLSL];
   let Args = [StringArgument<"Slot">, StringArgument<"Space", 1>];
   let Documentation = [HLSLResourceBindingDocs];
+  let AdditionalMembers = [{
+      enum class RegisterType : unsigned { SRV, UAV, CBuffer, Sampler, C, I, Invalid };
+
+      const FieldDecl *ResourceField = nullptr;
```

Comment:
Shouldn't this be made more abstract in the case of non-UDT resources? I would think the type would be Decl instead.

Reply:
I've removed the ResourceField.